README.md
<!--
 * @Author: Vane
 * @Date: 2021-08-19 19:08:17
 * @LastEditTime: 2021-09-16 14:04:18
 * @LastEditors: Vane
 * @Description:
 * @FilePath: \tp-cli\README.md
-->

# tp-cli（已更名为 vane-cli）

# 介绍

前端通用脚手架, 致力于提升前端研发效率和规范的工程化解决方案

随着业务项目越来越多，单纯的官方 cli 已经不足以满足定制的需要，我们需要一个囊括了已有项目公共组件、开箱即用的工程化项目模板，因此有了这个脚手架

通过该脚手架，你将拥有一支强大的基础项目模板军队，伴你在业务的战场中冲锋陷阵。

# 安装使用

1. 一键安装

```bash
npx vane-cli create
```

2. 全局安装

```bash
npm install -g vane-cli
```

```bash
vane-cli create
```

# 查看版本

```bash
vane --version
1.0.0
```

# 发布

```bash
# 设置镜像源
## 淘宝镜像
## npm config set registry https://registry.npm.taobao.org/
# 目前发布在npm官方仓库
npm config set registry https://registry.npmjs.org

npm login

npm run pub
```
**此脚本采取 `standard-version` 版本自增方式，发布之后及时 `git push` 本地代码。**

# 调试

此脚本已支持本地调试.

# 项目流程

✨✨✨[我是流程图](https://raw.githubusercontent.com/hudiegu/cdn/master/2022-05-25/10:46-4f8vB5.jpg)
![](https://raw.githubusercontent.com/hudiegu/cdn/master/2022-05-25/10:46-4f8vB5.jpg)


# 项目结构

```bash


├─ CHANGELOG.md —————————————————— 提交日志
├─ README.md ————————————————————— 项目介绍
├─ bin
│    └─ cli.js ——————————————————— 命令入口
├─ commitlint.config.js —————————— git提交规范
├─ global.d.ts ——————————————————— 类型声明
├─ package.json
├─ publish.sh ———————————————————— 发布脚本
├─ src
│    ├─ assets ——————————————————— 静态资源
│    │    └─ config.json ————————— 脚手架交互字典，后续迁至服务器
│    ├─ commands
│    │    ├─ create.ts ——————————— 创建项目模板
│    │    └─ index.ts ———————————— 创建入口
│    ├─ config
│    │    └─ constants.ts ———————— 全局配置
│    ├─ index.ts  ———————————————— 脚本入口
│    └─ utils
│           ├─ common.ts ————————— 公共函数
│           ├─ promps.ts ————————— cmd交互
│           ├─ rc.ts ————————————— 本地存储
│           └─ upgrade.ts ———————— 版本升级
├─ tsconfig.json ————————————————— ts
└─ yarn.lock


```

# 代码质量

ESLint + prettier 配合 husky 和 lint-staged，在代码提交时自动校验和修复代码格式


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.9.3](https://github.com/mottzz87/tp-cli/compare/v1.9.2...v1.9.3) (2023-05-18)


### Docs

* 更新readme ([b554449](https://github.com/mottzz87/tp-cli/commit/b554449525a294a5804050ee26fa441c89d9844f))
* 更新readme2 ([ac31318](https://github.com/mottzz87/tp-cli/commit/ac3131892f3cce80e55f0fc5ea8cf78b66b194a4))

### [1.9.2](https://github.com/mottzz87/tp-cli/compare/v1.9.1...v1.9.2) (2023-05-18)

### [1.9.1](https://github.com/mottzz87/tp-cli/compare/v1.9.0...v1.9.1) (2023-05-18)


### Docs

* 更新readme ([03d928c](https://github.com/mottzz87/tp-cli/commit/03d928cd18f713283de59889f09a589b450e6245))

## [1.9.0](https://github.com/mottzz87/tp-cli/compare/v1.8.2...v1.9.0) (2023-05-18)


### Features

* 新增log、基础常量 ([677658a](https://github.com/mottzz87/tp-cli/commit/677658abfe5a6911b53c857b331f301570063491))

### [1.8.2](https://github.com/mottzz87/tp-cli/compare/v1.8.1...v1.8.2) (2023-05-18)

### [1.8.1](https://github.com/mottzz87/tp-cli/compare/v1.8.0...v1.8.1) (2023-05-18)

## [1.8.0](https://github.com/mottzz87/tp-cli/compare/v1.7.3...v1.8.0) (2023-05-18)


### Features

* 去掉版本自增 ([0f86848](https://github.com/mottzz87/tp-cli/commit/0f868483445113b30fa46c6c7249c93dbb03fd92))

### [1.7.3](https://github.com/mottzz87/tp-cli/compare/v1.7.2...v1.7.3) (2023-05-17)

### [1.7.2](https://github.com/mottzz87/tp-cli/compare/v1.7.1...v1.7.2) (2023-05-17)

### [1.7.1](https://github.com/mottzz87/tp-cli/compare/v1.7.0...v1.7.1) (2023-05-17)

## [1.7.0](https://github.com/mottzz87/tp-cli/compare/v1.6.0...v1.7.0) (2023-05-17)


### Features

* 增加changelog日志类型 ([0f7afdd](https://github.com/mottzz87/tp-cli/commit/0f7afdd768a02d777874feb369349014f564aa38))
* 更新流程图 ([d47efae](https://github.com/mottzz87/tp-cli/commit/d47efae491078fe723d17c6636f93a33a46c4ca0))

## [1.6.0](https://github.com/mottzz87/tp-cli/compare/v1.5.1...v1.6.0) (2023-05-17)


### Features

* log ([1c094fb](https://github.com/mottzz87/tp-cli/commit/1c094fbd29685bc81029cc487eb5b9c32d9176af))

### [1.5.1](https://github.com/mottzz87/tp-cli/compare/v1.5.0...v1.5.1) (2023-05-17)

## [1.5.0](https://github.com/mottzz87/tp-cli/compare/v1.4.0...v1.5.0) (2023-05-17)


### Features

* log ([48a39ca](https://github.com/mottzz87/tp-cli/commit/48a39ca019795b64228b2e42a3d6e846c173bed2))

## [1.4.0](https://github.com/mottzz87/tp-cli/compare/v1.1.1...v1.4.0) (2022-05-25)


### Features

* 更新流程图 ([0e34d17](https://github.com/mottzz87/tp-cli/commit/0e34d178af37ee8cd3a3d0f4728317b603ead8ef))

### [1.1.1](https://github.com/mottzz87/tp-cli/compare/v1.1.0...v1.1.1) (2022-05-25)

## 1.1.0 (2022-05-25)


### Features

* 更新配置文件 ([1193654](https://github.com/mottzz87/tp-cli/commit/1193654bd7e49fc42fc4e5b117f9a302f2dee4d6))


src/index.ts
/*
 * @Author: Vane
 * @Date: 2021-08-19 19:06:06
 * @LastEditTime: 2021-09-08 13:07:26
 * @LastEditors: Vane
 * @Description: 入口
 * @FilePath: \tp-cli\src\index.ts
 */
import { program } from 'commander';
import chalk from 'chalk';
import { version, description } from '../package.json';
import Rc from './utils/rc';
import { GITLAB_URL, COMMAND_ALIAS, TEAM_LOGO, KEY_GITLAB_URL } from './config/constants';
import { upgrade } from './utils/upgrade';
import { printTeam, handleNoAuth, IOptions } from './utils/common';
import { create } from './commands';

program.version(version).description(description);

// create app-name
program
  .command('create')
  .description('创建项目，提供初始化项目模版选择')
  .option('-p, --projectName [projectName]', '项目名称')
  .option('-t, --type [H5 | PC | MINIAPP]', '项目类型')
  .option('-f, --frame [Vue | Taro | React]', '技术栈类型')
  .option('-au, --author', '作者')
  .option('-v, --version', '版本号')
  .option('-f, --force', '若目录存在则直接覆盖')
  .action(async (options: IOptions) => {
    // 逼格plus
    printTeam(TEAM_LOGO);

    // 检测升级
    await upgrade();

    // 无授权自动退出
    await handleNoAuth();

    //创建
    create(options);
  });

// 配置gitlab 本地存储
program
  .command('config')
  .description('录入脚手架gitlab配置信息')
  .action(() => {
    const args = process.argv.slice(3);
    const [action, key, value] = args;
    if (!args.length || !Object.keys(Rc).includes(action)) {
      console.log(chalk.redBright('🙄 Command input error, please refer to the following example command'));
      console.log('\nExamples:');
      console.log(chalk.gray('# Set configuration data'));
      console.log(chalk.yellow(`$ ${COMMAND_ALIAS} config set ${KEY_GITLAB_URL} ${GITLAB_URL}`));
      console.log(chalk.gray('# Read the specified configuration data'));
      console.log(chalk.yellow(`$ ${COMMAND_ALIAS} config get ${KEY_GITLAB_URL}`));
      console.log(chalk.gray('# Remove specified configuration data'));
      console.log(chalk.yellow(`$ ${COMMAND_ALIAS} config remove ${KEY_GITLAB_URL}`));
      console.log(chalk.gray('# View all configurations'));
      console.log(chalk.yellow(`$ ${COMMAND_ALIAS} config get`));
    } else {
      Rc[action](key, value);
    }
  })
  .on('--help', function () {
    console.log('\nExamples:');
    console.log(chalk.gray('# Set configuration data'));
    console.log(chalk.yellow(`$ ${COMMAND_ALIAS} config set ${KEY_GITLAB_URL} ${GITLAB_URL}`));
    console.log(chalk.gray('# Read the specified configuration data'));
    console.log(chalk.yellow(`$ ${COMMAND_ALIAS} config get ${KEY_GITLAB_URL}`));
    console.log(chalk.gray('# Remove specified configuration data'));
    console.log(chalk.yellow(`$ ${COMMAND_ALIAS} config remove ${KEY_GITLAB_URL}`));
    console.log(chalk.gray('# View all configurations'));
    console.log(chalk.yellow(`$ ${COMMAND_ALIAS} config list`));
  });

program.parse(process.argv);
